runtime/src/commonTest/kotlin/pbandk/json/JsonTest.kt

```diff
@@ -18,6 +19,40 @@ class JsonTest {
         assertContentEquals(byteArrayOf(), barBytes, "binary serialization should be empty for null field")
     }
 
+    @Test
+    fun testMessageWithEnumProto2() {
+        val jsonConfig = JsonConfig.DEFAULT.copy(compactOutput = true)
+
+        // This behavior is unexpected and will be fixed in a follow-up.
+        // See https://github.com/streem/pbandk/issues/235 for more details.
+        val message = MessageWithEnum()
+        assertEquals("{\"value\":null}", message.encodeToJsonString(jsonConfig))
```

Comment:
FWIW, this is compliant with the spec since `null` and not-present values in the JSON are supposed to be treated equivalently in the protobuf JSON encoding. But I agree that it'd be better if the `null` value is not output unless `JsonConfig.outputDefaultValues=true`.

Reply:
Agreed that the `null` doesn't hurt. I decided to remove the comments.

Reply:
Out of curiosity I tested official c++, python and java to see if they will output a `null` value in this scenario (by default outputting default values is set to false):
https://github.com/protocolbuffers/protobuf/pull/10812

The conclusion is that they don't emit an explicit `null`.